src/main/java/pl/marcin/stockmanagerapi/config/JwtAuthConverter.java

```diff
@@ -58,9 +61,12 @@ private Collection<? extends GrantedAuthority> extractResourceRoles(Jwt jwt) {
         }
         resource = (Map<String, Object>) resourceAccess.get(resourceId);
         resourceRoles = (Collection<String>) resource.get("roles");
-        return resourceRoles
+
+        var grantedAuthorities = resourceRoles
                 .stream()
-                .map(role -> new SimpleGrantedAuthority("ROLE_" + role))
+                .map(SimpleGrantedAuthority::new)
                 .collect(Collectors.toSet());
+
+        return roleHierarchy.getReachableGrantedAuthorities(grantedAuthorities);
```

Comment:
Here is the trick part to make roleHierarchy to work, we need to use this method so it'll add all the roles from the hierarchy configuration, for example, if we have ROLE_ADMIN it'll add the ROLE_STAFF automatically on the list.